toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Parser for the image builder's Networks configuration schema.
+
+package configuration
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"path/filepath"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/file"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+type Network struct {
+	BootProto   string   `json:"BootProto"`
+	GateWay     string   `json:"GateWay"`
+	Ip          string   `json:"Ip"`
+	NetMask     string   `json:"NetMask"`
+	OnBoot      bool     `json:"OnBoot"`
+	NameServers []string `json:"NameServer"`
+	Device      string   `json:"Device"`
+}
+
+// valid network boot protocols supported
+var validBootProtos = map[string]bool{
+	"dhcp":   true,
+	"static": true,
+	"bootp":  true,
+	"ibft":   true,
+	"":       true,
+}
+
+// UnmarshalJSON Unmarshals a Network entry
+func (n *Network) UnmarshalJSON(b []byte) (err error) {
+	// Use an intermediate type which will use the default JSON unmarshal implementation
+	type IntermediateTypeNetwork Network
+	err = json.Unmarshal(b, (*IntermediateTypeNetwork)(n))
+	if err != nil {
+		return fmt.Errorf("failed to parse [Network]: %w", err)
+	}
+
+	// Now validate the resulting unmarshaled object
+	err = n.IsValid()
+	if err != nil {
+		return fmt.Errorf("failed to parse [Network]: %w", err)
+	}
+	return
+}
+
+// IsValid returns an error if the Network struct is not valid
+func (n *Network) IsValid() (err error) {
+	err = n.bootProtoIsValid()
+	if err != nil {
+		return
+	}
+
+	err = n.ipAddrIsValid()
+	if err != nil {
+		return
+	}
+
+	err = n.deviceIsValid()
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+// ConfigureNetwork performs network configuration during the kickstart unattended installation process
+func ConfigureNetwork(installChroot *safechroot.Chroot, systemConfig SystemConfig) (err error) {
+	const squashErrors = false
+	var dnsUpdate bool
+
+	for _, networkData := range systemConfig.Networks {
+		deviceName, err := checkNetworkDeviceAvailability(networkData)
+		if err != nil {
+			return err
+		} else if deviceName == "" {
+			err = fmt.Errorf("The input network device is currently not supported on this system")
+			return err
+		}
+
+		err = createNetworkConfigFile(installChroot, networkData, deviceName)
+		if err != nil {
+			return err
+		}
+
+		dnsUpdate = len(networkData.NameServers) > 0
```

Comment:
>dnsUpdate = len(networkData.NameServers) > 0 [](http://example.com/codeflow?start=2&length=44)

There's something fishy here. Since we're in a loop, only the last `networkData` will have any effect on the final value of `dnfUpdate`.

Reply:
Good catch. Fixed the logic here.